testenv/gcp/gcp.go

```diff
@@ -235,6 +240,13 @@ func New(config *Config, opts ...ConfigOptFn) (*GCPEnv, error) {
 			return nil, fmt.Errorf("failed to generate ssh key pair: %w", err)
 		}
 	}
+	privateKeyFile := filepath.Join(config.WorkDir, "id_rsa")
+	publicKeyFile := filepath.Join(config.WorkDir, "id_rsa.pub")
+	if _, err := os.Stat(privateKeyFile); errors.Is(err, os.ErrNotExist) {
+		if err := sshKeyPair.Save(privateKeyFile, publicKeyFile); err != nil {
+			return nil, fmt.Errorf("failed to save SSH keys to filesystem: %w", err)
+		}
+	}
```

Comment:
seems like this part of the code (+ssh generation) is common for all e2e test providers. Is it possible to move the shared SSH code to `testenv/utils/ssh.go`?

Reply:
Good idea, created a function for this common step!